### PR TITLE
Remove GetNodeData

### PR DESCRIPTION
**Pull request by rakita**
- Remove all functions that GetNodeData used

### CODE REVIEW

**Comment by rakita** [comment]
Will wait for ipfs and light client removal to be merged before this comes. There are some dependencies on those two modules.

File: ethcore/src/json_tests/skip.rs [hunk 1]
```diff
@@ -28,5 +28,6 @@ lazy_static! {
 
 #[cfg(not(feature = "ci-skip-tests"))]
 lazy_static! {
-    pub static ref SKIP_TEST_STATE: ethjson::test::SkipStates = ethjson::test::SkipStates::empty();
+    pub static ref SKIP_TEST_STATE: ethjson::test::SkipStates =
+        { ethjson::test::SkipStates::empty() };
```
Comment (vorot93) [inline ethcore/src/json_tests/skip.rs:32]: revert

File: ethcore/src/client/client.rs [hunk 2]
```diff
@@ -91,8 +91,10 @@ use trace::{
 use transaction_ext::Transaction;
 use verification::{
     self,
-    queue::kind::{blocks::Unverified, BlockLike},
-    BlockQueue, PreverifiedBlock, Verifier,
+    {
+        queue::kind::{blocks::Unverified, BlockLike},
+        BlockQueue, PreverifiedBlock, Verifier,
+    },
```
Comment (vorot93) [inline ethcore/src/client/client.rs:97]: revert

File: ethcore/blockchain/src/blockchain.rs [hunk 3]
```diff
@@ -650,40 +650,42 @@ impl BlockChain {
...
+                batch.put(db::COL_EXTRA, b"best", &hash);
+                bc.db.key_value().write(batch).expect(
                     "Low level database error when fetching 'best' block. Some issue with disk?",
                 );
-                    hash
-                }
-            };
+                hash
+            }
+        };
```
Comment (vorot93) [inline ethcore/blockchain/src/blockchain.rs:688]: rebase leftover? revert?

File: accounts/ethstore/src/presale.rs [hunk 4]
```diff
@@ -14,7 +14,9 @@
 // You should have received a copy of the GNU General Public License
 // along with Parity Ethereum.  If not, see <http://www.gnu.org/licenses/>.
 
-use crypto::{self, pbkdf2, Keccak256};
+use crypto::{
+    self, {pbkdf2, Keccak256},
+};
```
Comment (vorot93) [inline accounts/ethstore/src/presale.rs:19]: revert

File: ethcore/light/src/provider.rs [hunk 5]
```diff
@@ -246,9 +246,10 @@ impl<T: ProvingBlockChainClient + ?Sized> Provider for T {
...
     }
 
-    fn contract_code(&self, req: request::CompleteCodeRequest) -> Option<request::CodeResponse> {
-        self.state_data(&req.code_hash)
-            .map(|code| ::request::CodeResponse { code })
+    fn contract_code(&self, _req: request::CompleteCodeRequest) -> Option<request::CodeResponse> {
+        None
+        //self.state_data(&req.code_hash)
+        //	.map(|code| ::request::CodeResponse { code })
     }
```
Comment (vorot93) [inline ethcore/light/src/provider.rs:253]: remove

File: ethcore/sync/src/chain/supplier.rs [hunk 6]
```diff
@@ -253,35 +253,8 @@ impl SyncSupplier {
...
-        }
-        trace!(target: "sync", "{} -> GetNodeData: return {} entries", peer_id, added);
-        let mut rlp = RlpStream::new_list(added);
-        for d in data {
-            rlp.append(&d);
-        }
-        Ok(Some((NodeDataPacket.id(), rlp)))
+    fn return_empty(_io: &dyn SyncIo, _r: &Rlp, _peer_id: PeerId) -> RlpResponseResult {
+        return Ok(None);
     }
```
Comment (vorot93) [inline ethcore/sync/src/chain/supplier.rs:258]: remove

File: ipfs/src/route.rs [hunk 7]
```diff
@@ -105,23 +105,19 @@ impl IpfsHandler {
...
-            .client()
-            .state_data(&hash)
-            .ok_or(Error::StateRootNotFound)?;
+    fn state_trie(&self, _hash: H256) -> Result<Out> {
+        Err(Error::ContractNotFound)
+        //let data = self.client().state_data(&hash).ok_or(Error::StateRootNotFound)?;
 
-        Ok(Out::OctetStream(data))
+        //Ok(Out::OctetStream(data))
     }
```
Comment (vorot93) [inline ipfs/src/route.rs:113]: remove?

File: ipfs/src/route.rs [hunk 8]
```diff
@@ -105,23 +105,19 @@ impl IpfsHandler {
...
-            .state_data(&hash)
-            .ok_or(Error::ContractNotFound)?;
+    fn contract_code(&self, _hash: H256) -> Result<Out> {
+        Err(Error::ContractNotFound)
+        // let data = self.client().state_data(&hash).ok_or(Error::ContractNotFound)?;
 
-        Ok(Out::OctetStream(data))
+        // Ok(Out::OctetStream(data))
     }
 }
```
Comment (vorot93) [inline ipfs/src/route.rs:122]: remove?

File: accounts/ethstore/src/presale.rs (hunk 4 above)
Comment (rakita) [reply accounts/ethstore/src/presale.rs:19]: This pull request is still not ready for review, I didn't add anyone to review list. I waited for ipfs/light client to be merged before I come to this and fix it.

**Comment by rakita** [comment]
Should be okay for review

File: ethcore/evm/src/lib.rs [hunk 9]
```diff
@@ -32,11 +32,12 @@ extern crate lazy_static;
 #[cfg_attr(feature = "evm-debug", macro_use)]
 extern crate log;
 
-#[cfg(test)]
-extern crate hex_literal;
 #[cfg(test)]
 extern crate rustc_hex;
 
+#[cfg(test)]
```
Comment (adria0) [inline ethcore/evm/src/lib.rs:38]: What about just discarding changes in this file?

**Pull request merged**